src/task_scl_structural_habitat.py

```diff
@@ -25,36 +25,344 @@ class str_hab(SCLTask):
         },
     }
 
-
-
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.land_cover, _ = self.get_most_recent_image(
             ee.ImageCollection(self.inputs["land_cover"]["ee_path"])
         )
         self.elevation = ee.Image(self.inputs["elevation"]["ee_path"])
         self.zones = ee.FeatureCollection(self.inputs["zones"]["ee_path"])
-        self.set_aoi_from_ee(self.inputs["probability"]["ee_path"])
-
+        # self.set_aoi_from_ee(self.inputs["probability"]["ee_path"])
 
     def calc(self):
 
-        print(land_cover)
+        zones_img = self.zones.reduceToImage(
+            properties=['Biome_zone'],
+            reducer=ee.Reducer.mode()
+        ).rename('Biome_zone')
+
+        def remap_elev(from_list, to_list, zone_number, z_img, elev_img, lc_img):
+            return (elev_img.lt(
+            lc_img.remap(from_list, to_list)
+                .updateMask(z_img.eq(zone_number)))
+                .multiply(100))
+
+
+        def remap_lc_by_zone(lc_list, z1_list, z2_list, z3_list, z_img, elev_img, lc_img):
+            lc_zone1=remap_elev(lc_list, z1_list, 1,
+                                zones_img, elev_img, lc_img)
+            lc_zone2=remap_elev(lc_list, z2_list, 2,
+                                zones_img, elev_img, lc_img)
+            lc_zone3=remap_elev(lc_list, z3_list, 3,
+                                zones_img, elev_img, lc_img)
+            return (ee.ImageCollection.fromImages([lc_zone1, lc_zone2, lc_zone3])
+                .mosaic()
+                .selfMask()
+                .rename('str_hab'))
+
+        lcClassification=[{
```

Comment:
What would it take to create this from a table, rather than hardcoding the vals in the code?